### PR TITLE
chore(ionic): update to Ionic v7 and Angular 15

### PR DESCRIPTION
**Pull request by brandyscarney**
- Updates Ionic to v7
- Updates Angular to v15
- Migrates to Cypress testing instead of Protractor
  - Tests are located in `e2e/specs/`
  - Fixtures are located in `e2e/fixtures/`
- Migrates swiper to scroll-snap
- Adds [stylelint](https://stylelint.io/) matching the config in the `ionic-framework` in order to clean up the scss files (this also runs with `npm run lint`

### CODE REVIEW

File: src/app/pages/schedule-filter/schedule-filter.html [hunk 1]
```diff
@@ -19,11 +19,12 @@
...
-      <ion-icon *ngIf="ios" slot="start" [name]="track.icon" color="medium"></ion-icon>
-      <ion-label>{{track.name}}</ion-label>
-      <ion-checkbox [(ngModel)]="track.isChecked"></ion-checkbox>
-    </ion-item>
+<ion-item *ngFor="let track of tracks" [attr.track]="track.name | lowercase">
+  <ion-icon *ngIf="ios" slot="start" [name]="track.icon" color="medium"></ion-icon>
+  <ion-checkbox [(ngModel)]="track.isChecked">
+    {{track.name}}
+  </ion-checkbox>
+</ion-item>
```
Comment (amandaejohnston) [inline src/app/pages/schedule-filter/schedule-filter.html:27]: Looks like opening the filter box a second time causes it to use legacy checkboxes again for some reason?

https://user-images.githubusercontent.com/90629384/228283907-44284394-7815-4ee3-a1ce-bd1e06ba2d88.mp4

Comment (amandaejohnston) [inline src/app/pages/schedule-filter/schedule-filter.html:27]: The filters also no longer seem to be working. On the `master` branch, I can uncheck categories and it'll actually filter those out, but all events always display when on the branch.

File: src/app/pages/tutorial/tutorial.scss [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
 .slide-container {
+	display: grid;
+	grid-template-columns: repeat(4, 100%);
+	grid-template-rows: 1fr;
+
+	height: 100%;
+
+  overflow: scroll;
+	scroll-snap-type: x mandatory;
+}
```
Comment (amandaejohnston) [inline src/app/pages/tutorial/tutorial.scss:18]: It looks like this spacing is caused by some of the lines using tabs and others using spaces; could we get those consistent?

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

File: src/app/pages/schedule-filter/schedule-filter.html (hunk 1 above)
Comment (sean-perkins) [reply src/app/pages/schedule-filter/schedule-filter.html:27]: This looks to possibly be a bug with the modern syntax + Angular. I'm not able to get the same behavior with the modern syntax and the React conference app. However the React conference app is using inline modals and the Angular conference app is using controller modals. 

Adding `aria-label` to the `ion-checkbox` does consistently render correctly, but for some reason our automatic detection of a slotted label is not working in Angular (when rendered the second+ time).

e.g.: This "works around" the issue.
```html
<ion-checkbox [(ngModel)]="track.isChecked" [attr.aria-label]="track.name">
   {{ track.name }}
</ion-checkbox>
```

Comment (brandyscarney) [reply src/app/pages/schedule-filter/schedule-filter.html:27]: This was caused by [removing the `hidden` attribute CSS](https://github.com/ionic-team/ionic-framework/blob/feature-7.0/BREAKING.md#hidden-attribute). Should we move this into the migration guide? It seems like something that is easy to overlook.

Comment (liamdebeasi) [reply src/app/pages/schedule-filter/schedule-filter.html:27]: The migration guide should be focused on changes that are required in order for your app to compile. Perhaps we could move this change to the top of the file?

Comment (amandaejohnston) [reply src/app/pages/schedule-filter/schedule-filter.html:27]: I wouldn't be opposed to it 👀

Comment (liamdebeasi) [reply src/app/pages/schedule-filter/schedule-filter.html:27]: Sure CSS Utilities works - want to open a PR?

Comment (brandyscarney) [reply src/app/pages/schedule-filter/schedule-filter.html:27]: Yeah I can do that!

File: src/app/pages/tutorial/tutorial.scss [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
 
 .slide-container {
+	display: grid;
+	grid-template-columns: repeat(4, 100%);
+	grid-template-rows: 1fr;
+
+	height: 100%;
+
+  overflow: scroll;
+	scroll-snap-type: x mandatory;
```
Comment (liamdebeasi) [inline src/app/pages/tutorial/tutorial.scss:17]: Can we add `scroll-snap-stop` you can't accidentally swipe over multiple slides: https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-stop

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Can we also evaluate `ion-input` and `ion-textarea` usages to see if we need to listen for `ionInput` instead of `ionChange`?

File: src/app/pages/tutorial/tutorial.scss (hunk 2 above)
Comment (brandyscarney) [reply src/app/pages/tutorial/tutorial.scss]: Added stylelint to autocorrect this and other style issues based on the framework's config: https://github.com/ionic-team/ionic-conference-app/pull/1790/commits/e6c9d27ac644898bbc0d36433ccbd436cf1535d0

File: src/app/pages/schedule-filter/schedule-filter.html (hunk 1 above)
Comment (brandyscarney) [reply src/app/pages/schedule-filter/schedule-filter.html:27]: I confirmed that this only happens on the new syntax. It is loading from this file: https://github.com/ionic-team/ionic-conference-app/blob/master/src/assets/data/data.json

```typescript
return this.http
  .get('assets/data/data.json')
  .pipe(map(this.processData, this));
```

**Review by liamdebeasi — APPROVED** [review]

File: src/app/pages/schedule-filter/schedule-filter.html (hunk 1 above)
Comment (brandyscarney) [reply src/app/pages/schedule-filter/schedule-filter.html]: I tried what @sean-perkins mentioned on the call this morning and do not see the same problem with `ion-toggle` so it seems to only affect `ion-checkbox`. 

I added `aria-label` as a temporary workaround and created a ticket to fix this in the framework - [FW-3891](https://ionic-cloud.atlassian.net/browse/FW-3891). I referenced the ticket above in the commit with the workaround - https://github.com/ionic-team/ionic-conference-app/pull/1790/commits/19110292e038b941eb57d093f689e42175cc405f.

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**